src/style/style_layer/heatmap_style_layer.js

```diff
@@ -58,12 +67,23 @@ class HeatmapStyleLayer extends StyleLayer {
         }
     }
 
-    queryRadius(): number {
-        return 0;
+    queryRadius(bucket: Bucket): number {
+        return getMaximumPaintValue('heatmap-radius', this, ((bucket: any): CircleBucket<*>));
     }
 
-    queryIntersectsFeature(): boolean  {
-        return false;
+    queryIntersectsFeature(queryGeometry: TilespaceQueryGeometry,
+                           feature: VectorTileFeature,
+                           featureState: FeatureState,
+                           geometry: Array<Array<Point>>,
+                           zoom: number,
+                           transform: Transform,
+                           pixelPosMatrix: Float32Array,
+                           elevationHelper: ?DEMSampler): boolean {
+
+        const size = this.paint.get('heatmap-radius').evaluate(feature, featureState);
+        return queryIntersectsCircle(
+            queryGeometry, geometry, transform, pixelPosMatrix, elevationHelper,
+            true, true, new Point(0, 0), size);
```

Comment:
TIL we don't have `heatmap-translate` I thought it was feature available across all layers.